presto-native-execution/scripts/release-centos-dockerfile/opt/entrypoint.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+set -eExv -o functrace
+
+SCRIPT_DIR=$(readlink -f "$(dirname "${BASH_SOURCE[0]}")")
+PRESTO_HOME="${PRESTO_HOME:-"/opt/presto/"}"
+USE_ENV_PARAMS=${USE_ENV_PARAMS:-0}
+
+source "${SCRIPT_DIR}/common.sh"
+
+trap 'exit 2' SIGSTOP SIGINT SIGTERM SIGQUIT
+trap 'failure "LINENO" "BASH_LINENO" "${BASH_COMMAND}" "${?}"; [ -z "${DEBUG}" ] && exit 1 || sleep 3600' ERR
+
+if [[ "${DEBUG}" == "0" || "${DEBUG}" == "false" || "${DEBUG}" == "False" ]]; then DEBUG=""; fi
+
+http_server_port=8080
+discovery_uri="http://127.0.0.1:${http_server_port}"
+
+while getopts ':-:' optchar; do
+  case "$optchar" in
+    -)
+      case "$OPTARG" in
+        discovery-uri=*) discovery_uri="${OPTARG#*=}" ;;
+        http-server-port=*) http_server_port="${OPTARG#*=}" ;;
+        use-env-params) USE_ENV_PARAMS=1 ;;
+        *)
+          presto_args+=($optchar)
+          ;;
+      esac
+      ;;
+    *)
+      presto_args+=($optchar)
+      ;;
+  esac
+done
+
+function node_command_line_config()
+{
+  printf "presto.version=0.273.3\n"                    >  "${PRESTO_HOME}/config.properties"
+  printf "discovery.uri=${discovery_uri}\n"            >> "${PRESTO_HOME}/config.properties"
+  printf "http-server.http.port=${http_server_port}\n" >> "${PRESTO_HOME}/config.properties"
+
+  printf "node.environment=intel-poland\n" >  "${PRESTO_HOME}/node.properties"
+  printf "node.location=torun-cluster\n"  >> "${PRESTO_HOME}/node.properties"
+  printf "node.id=${NODE_UUID}\n"        >> "${PRESTO_HOME}/node.properties"
+  printf "node.ip=$(hostname -I)\n"      >> "${PRESTO_HOME}/node.properties"
+}
+
+function node_configuration()
+{
+  cat "${PRESTO_HOME}/config.properties.template" > "${PRESTO_HOME}/config.properties"
```

Comment:
is this expected to be mounted as a volume when running the docker container? Should we copy the default ones on the Dockerfile?
If I try to do a docker run it fails to find it:
```
$ docker run presto/prestissimo-avx-centos:latest
...
cat /opt/presto//config.properties.template
cat: /opt/presto//config.properties.template: No such file or directory
```
If I override the entrypoint to see if the file is there is missing:
```
$ docker run --entrypoint /bin/bash -it presto/prestissimo-avx-centos:latest
bash-4.4$ ls /opt/presto/
_repo  catalog	common	config.properties  connectors  etc  http  libpresto_server_lib.a  operators  presto_server  tests  thrift  types
bash-4.4$ cat /opt/presto/config.properties 
bash-4.4$ 
```
Should we add something on the README on how to run it after built using docker?

Reply:
node_configuration is designed for use with kubernetes and mounting configuration files as templates that are latet in runtime filled with node-specific values and save as desired configuration files.

Reply:
To run it using docker one should use something like:
`docker run presto/prestissimo-avx-centos:latest --use-env-params --discovery-uri=http://presto-coordinaator.default.svc.cluster.local:8080 --http-server-port=8080`

Reply:
Hi @Mionsz the work on this PR is great!
Maybe as a new user I would like to have a really simple guide with the two default commands to run locally with docker:
- build --> this is covered with the `make runtime-container` command
- run --> I would like to have a default command saying if you run `docker run xxxxx` this will start a local instance with default settings exposing at port 8080 after building with `make runtime-container`

Reply:
I just saw your previous comment for:
`docker run presto/prestissimo-avx-centos:latest --use-env-params --discovery-uri=http://presto-coordinaator.default.svc.cluster.local:8080 --http-server-port=8080`
this works even though I am getting some failures, I'll investigate:
```
"${PRESTO_HOME}/presto_server" --logtostderr=1 --v=1 "${presto_args[@]}"
+ /opt/presto//presto_server --logtostderr=1 --v=1
I1108 12:57:27.067461    12 PrestoServer.cpp:332] Starting with node memory 40GB
E1108 12:57:27.068285    12 MmapAllocator.cpp:361] mmap failed with 12
E1108 12:57:27.068301    12 Exceptions.h:68] Line: ../../velox/velox/common/memory/MmapAllocator.cpp:362, Function:SizeClass, Expression:  Could not allocate working memorymmap failed with 12, Source: RUNTIME, ErrorCode: INVALID_STATE
E1108 12:57:27.068984    12 ExceptionTracer.cpp:218] terminate() called, exception stack follows
E1108 12:57:27.068993    12 ExceptionTracer.cpp:220] Exception type: facebook::velox::VeloxRuntimeError (11 frames)
    @ 0000000000b47eb5 _ZZN12_GLOBAL__N_111InitializerC4EvENKUlPvPSt9type_infoPPFvS1_EE_clES1_S3_S6_
                       /opt/dependency/folly/_build/../folly/experimental/exception_tracer/ExceptionStackTraceLib.cpp:79
    @ 0000000000b47ee5 _ZZN12_GLOBAL__N_111InitializerC4EvENUlPvPSt9type_infoPPFvS1_EE_4_FUNES1_S3_S6_
                       /opt/dependency/folly/_build/../folly/experimental/exception_tracer/ExceptionStackTraceLib.cpp:79
    @ 0000000000b4ee23 __cxa_throw
                       /opt/dependency/folly/_build/../folly/experimental/exception_tracer/ExceptionTracerLib.cpp:58
    @ 0000000003487376 _ZN8facebook5velox6detail14veloxCheckFailINS0_17VeloxRuntimeErrorERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEEvRKNS1_18VeloxCheckFailArgsET0_
    @ 000000000347c753 _ZN8facebook5velox6memory13MmapAllocator9SizeClassC2Emm
    @ 000000000347f80b _ZN8facebook5velox6memory13MmapAllocatorC1ERKNS1_20MmapAllocatorOptionsE
    @ 000000000067cd24 _ZN8facebook6presto12PrestoServer20initializeAsyncCacheEv
    @ 000000000067d668 _ZN8facebook6presto12PrestoServer3runEv
    @ 0000000000647104 main
    @ 000000000003ad85 __libc_start_main
    @ 00000000006513ee _start

E1108 12:57:27.103837    12 ExceptionTracer.cpp:222] exception stack complete
terminate called after throwing an instance of 'facebook::velox::VeloxRuntimeError'
  what():  Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: Could not allocate working memorymmap failed with 12
Retriable: False
Function: SizeClass
File: ../../velox/velox/common/memory/MmapAllocator.cpp
Line: 362
Stack trace:
# 0  _ZN8facebook5velox7process10StackTraceC1Ei
# 1  _ZN8facebook5velox14VeloxExceptionC2EPKcmS3_St17basic_string_viewIcSt11char_traitsIcEES7_S7_S7_bNS1_4TypeES7_
# 2  _ZN8facebook5velox6detail14veloxCheckFailINS0_17VeloxRuntimeErrorERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEEvRKNS1_18VeloxCheckFailArgsET0_
# 3  _ZN8facebook5velox6memory13MmapAllocator9SizeClassC2Emm
# 4  _ZN8facebook5velox6memory13MmapAllocatorC1ERKNS1_20MmapAllocatorOptionsE
# 5  _ZN8facebook6presto12PrestoServer20initializeAsyncCacheEv
# 6  _ZN8facebook6presto12PrestoServer3runEv
# 7  main
# 8  __libc_start_main
# 9  _start

*** Aborted at 1667908647 (Unix time, try 'date -d @1667908647') ***
*** Signal 6 (SIGABRT) (0xba0000000c) received by PID 12 (pthread TID 0x7fc7e3be0e40) (linux TID 12) (maybe from PID 12, UID 186) (code: -6), stack trace: ***
(error retrieving stack trace)
```

Reply:
Seems like it could not allocate memory, 40GB?

Reply:
Adding this in config.properties should help
```ini
query.max-memory-per-node=2
query.max-total-memory-per-node=3
```

Reply:
ok, could we add the ability to configure the memory allocated to be overridden too? This worked locally:
```
printf "node.memory_gb=${memory_gb}\n" >> "${PRESTO_HOME}/node.properties"
```

Reply:
This should be done by config file mounted inside the container. We should not override the defaults using scripting I think.

Reply:
It would be better to add info that default configuration requires 40GB of memory to be reserved by Prestissimo, as this is a configuration setting.